### ISSUE TITLE
Profile UDID (environment variable) not carried to gym from match

### ISSUE THREAD

**Issue by Rameshv** [fastlane/fastlane#5374]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I am using `match` as part of fastlane to download the certs & profiles during the build. The issue is match works perfect and downloads the correct certs & profiles and install it. But the UDID of the profile is not correctly passed to the next `gym` lane. This causes builds to fail with this error

```
❌  Code Sign error: No matching provisioning profiles found: No provisioning profiles with a valid signing identity (i.e. certificate and private key pair) matching the bundle id
entifier “com.xxxxxxxx.app” were found.
```

and

```
[20:17:20]: Exit status: 65
[20:17:20]: You don't have a provisioning profile for the bundle identifier 'com.xxxxxx.app' installed on the local machine
[20:17:20]: Make sure you have the profile on this computer and it's properly installed
[20:17:20]: You can use sigh (https://github.com/fastlane/fastlane/tree/master/sigh) to download and install the provisioning profile
[20:17:20]: Follow this guide: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/CodeSigning.md
```

Based on the match output, i am seeing this environment `sigh_com.xxxxx.app_adhoc ` variable set to correct UDID (you can see that in the below log). But when i print this env variable from the lane, it prints nothing

This is a new app, and i have all the certs/keys generated new in the same machine and all are installed properly. But gym seems not find the correct code signing certs.

Remember i dont use `cert/sigh` inside my lanes, assuming  `match` takes care of that (since its built on top of them). is that correct?

##### Complete output when running fastlane, including the stack trace and command used

```
+-----------------------+------------------------------------------+
|                     Summary for match 0.6.1                      |
+-----------------------+------------------------------------------+
| type                  | adhoc                                    |
| force_for_new_devices | true                                     |
| shallow_clone         | true                                     |
| git_url               | git@github.com/xxxxxxxx/ios-certs.git |
| app_identifier        | com.inject.app                           |
| username              | xxx@xxxx.xom                 |
| git_branch            | master                                   |
| keychain_name         | login.keychain                           |
| readonly              | false                                    |
| team_id               | xxxxxxxxx                               |
| verbose               | false                                    |
| force                 | false                                    |
| skip_docs             | false                                    |
+-----------------------+------------------------------------------+

[18:30:57]: Cloning remote git repo...
[18:31:02]: 🔓  Successfully decrypted certificates repo
[18:31:02]: Verifying that the certificate and profile are still valid on the Dev Portal...
[18:31:20]: Installing certificate...
[18:31:32]: Installing provisioning profile...

+----------------------+--------------------------------------+
|               Installed Provisioning Profile                |
+----------------------+--------------------------------------+
| App Identifier       | com.xxxxx.app                       |
| Type                 | adhoc                                |
| UUID                 | 11313-61688-4dad-89f4-11xxx |
| Environment Variable | sigh_com.xxxx.app_adhoc            |
+----------------------+--------------------------------------+

[18:31:34]: All required keys, certificates and provisioning profiles are installed 🙌
[18:31:34]: 🔒  Successfully encrypted certificates repo
[18:31:34]: Setting Provisioning Profile type to 'ad-hoc'
[18:31:35]: PROFILE_UDID is :: 
[18:31:35]: -----------------
[18:31:35]: --- Step: gym ---
[18:31:35]: -----------------


❌  Code Sign error: No matching provisioning profiles found: No provisioning profiles with a valid signing identity (i.e. certificate and private key pair) matching the bundle id
entifier “com.inject.app” were found.


[18:33:11]: Exit status: 65
[18:33:11]: You don't have a provisioning profile for the bundle identifier 'com.inject.app' installed on the local machine
[18:33:11]: Make sure you have the profile on this computer and it's properly installed
[18:33:11]: You can use sigh (https://github.com/fastlane/fastlane/tree/master/sigh) to download and install the provisioning profile
[18:33:11]: Follow this guide: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/CodeSigning.md

```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
platform :ios do
  before_all do |lane, options|
    cocoapods
    if options[:version] 
      increment_version_number(version_number: options[:version])
      increment_build_number(build_number: '1')
    else
      increment_build_number
    end
  end
desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do |options|
    match(type: "adhoc",force_for_new_devices: true,shallow_clone: true) # more information: https://codesigning.guide

    
    ENV["PROFILE_UDID"] = lane_context[SharedValues::SIGH_UDID]
    # ENV["PROFILE_UDID"] = ENV["sigh_com.xxxxx.app_adhoc"]
   
    puts "PROFILE_UDID is :: #{ENV["PROFILE_UDID.xxxxx.app_adhoc"]}"
    gym(scheme:"INJECT",
      export_method: "ad-hoc",
      output_directory: "./build")

    clean_build_artifacts

    hockey_it

    pilot(skip_waiting_for_build_processing: true)

    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end
end
```

### Environment

fastlane version (run `fastlane -v`): 1.97.2

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)?nope

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?yep, rvm

**Comment by Rameshv** [comment]
the problem is `sigh_com.xxxx.app_adhoc` is not configured properly. It works after correct env variable

**Comment by PitchLabsAsh** [comment]
Did this get resolved?

I have seen similar issues many times on our build machine useing

`match(app_identifier: "com.testapp.public", type: "adhoc", readonly: true)`

If I update the profile from another machine to include new devices match will correctly fetch the new profile but the env_var does not seem to update in some way as the build ends up being signed with the previous profile.

I seem to have to reboot or manually go and delete old profiles in order to get the build process to realise that its changed.

**Comment by TKBurner** [comment]
@PitchLabsAsh Does this happen if you add the force: true parameter to your match command?

**Comment by PitchLabsAsh** [comment]
Yes it still happens with force: true.

See issue https://github.com/fastlane/fastlane/issues/1393

**Comment by grioja** [comment]
@TKBurner having the same problem as @PitchLabsAsh 

The problem is that xcode builds using a name of the profile now, which is bad because everytime match regenerates a profile (and uses the same name), the old profiles remain. So you end up with several old profiles alongside the new profile, and since it goes by name... it does not pick the newest profile, and thus exports via gym with the "wrong" provisioning profile.

Solutions that I can see are to clean out old profiles if match updates to a new one, or to be able to specify the profile by UDID

**Comment by TKBurner** [comment]
@grioja Have you tried using the `provisioning_profile_specifier`?

**Comment by grioja** [comment]
@TKBurner yes we use that to specify the profile name.. but the problem is that it is too vague for xcode...

this was tricky to track down in our CI process... we did a manual export ourselves via xcode and then upload to crashlytics and it worked for our new devices... but everything we tried gym will not export with the correct profile.

**Comment by grioja** [comment]
I think xcode walks the directory (which is sorted alphabetically by UDID) and finds the first match given the provisioning_profile_specifier name.

We need some way to clear out the "old profiles" to rid ourselves of false positive profile matches.

**Comment by grioja** [comment]
@TKBurner confirmed this behavior: we wiped out all our profiles on our build machine and ran a clean build (with a match adhoc readonly) and the export with a fresh pull of the profiles works. So there is absolutely a problem with old profiles being left around that confuses xcodebuild

**Comment by PitchLabsAsh** [comment]
I'm having less problems with this now.

Xcode does use the name but the documentation I'm sure claims that it uses the  most recent if there are duplicates.

The Xcode project now has both a provisioning profile and "deprecated" provisioning profile setting. I found I had to fully delete the deprecated setting and make sure only the new field was set, after that Xcode seemed to use the correct profile for me.

Ash
